android/quest/src/main/java/org/smartregister/fhircore/quest/data/patient/PatientRegisterPagingSource.kt

```diff
@@ -46,30 +45,30 @@ class PatientRegisterPagingSource(
    *
    * nextKey = if (data.isNotEmpty()) pageNumber + 1 else null
    */
-  override suspend fun load(params: LoadParams<Int>): LoadResult<Int, RegisterViewData> {
+  override suspend fun load(params: LoadParams<Int>): LoadResult<Int, ResourceData> {
     return try {
       val currentPage = params.key ?: _patientPagingSourceState.currentPage
-      val data =
+      val registerData =
         patientRegisterRepository.loadRegisterData(
-            currentPage = currentPage,
-            appFeatureName = _patientPagingSourceState.appFeatureName,
-            healthModule = _patientPagingSourceState.healthModule,
-            loadAll = _patientPagingSourceState.loadAll
-          )
-          .map { registerViewDataMapper.transformInputToOutputModel(it) }
+          currentPage = currentPage,
+          registerId = _patientPagingSourceState.registerId
+        )
+
       val prevKey =
         when {
           _patientPagingSourceState.loadAll -> if (currentPage == 0) null else currentPage - 1
           else -> null
         }
       val nextKey =
         when {
-          _patientPagingSourceState.loadAll -> if (data.isNotEmpty()) currentPage + 1 else null
+          _patientPagingSourceState.loadAll ->
+            if (registerData.isNotEmpty()) currentPage + 1 else null
           else -> null
         }
 
-      LoadResult.Page(data = data, prevKey = prevKey, nextKey = nextKey)
+      LoadResult.Page(data = registerData, prevKey = prevKey, nextKey = nextKey)
     } catch (exception: Exception) {
```

Comment:
I understand this is existing code, but I'm always suspicious when I see a blank Exception catch instead of a more specific exception class